### PR TITLE
finish react redux advanced demo

### PR DESCRIPTION
**Pull request by mvillafuertem**

### CODE REVIEW

File: react-redux/src/main/scala/demo/advanced/ExpenseContainer.scala
Comment (oyvindberg) [inline react-redux/src/main/scala/demo/advanced/ExpenseContainer.scala]: is it possible to scalafy this a bit? Something like
```scala

trait LinkDispatchProps { // extends js.Object if it has to
  def startAddExpense(e: Expense): Unit
  // ...
}

new LinkDispatchProps { 
  override def startAddExpense(e: Expense): Unit = dispatch(setExpenses(e))
}
```

Comment (mvillafuertem) [reply react-redux/src/main/scala/demo/advanced/ExpenseContainer.scala]: I'm going to try it

Comment (mvillafuertem) [reply react-redux/src/main/scala/demo/advanced/ExpenseContainer.scala]: I get this error :(

`Uncaught TypeError: Cannot read property 'dispatch$1' of undefined`

Comment (oyvindberg) [reply react-redux/src/main/scala/demo/advanced/ExpenseContainer.scala]: There is both a commit and this comment, does that mean you got it to work in the end?

Comment (mvillafuertem) [reply react-redux/src/main/scala/demo/advanced/ExpenseContainer.scala]: Yes, I was working on this. I think it looks better now

**Comment by oyvindberg** [comment]
Hey I had a look at this now. I applied some of my old redux skills and made it a bit more modular and less boilerplatey, removed action creators and some stuff like that I never saw the point with in a typed language.

I still think it's a bad idea to use redux with Scala.js given the very hard constraints it puts on action objects, but at least this way I can see it's possible. Have a look at https://github.com/ScalablyTyped/ScalaJsReactDemos/tree/redux-redux

**Comment by mvillafuertem** [comment]
Wow!! 😱 The example really looks great. Thanks

**Review by oyvindberg — APPROVED** [review]
Could you pick up my commit and rebase on master?

I just gave you privileges so I think you can go ahead and merge this when you're happy with it

**Comment by mvillafuertem** [comment]
Done!

I think the invitation is expired because I can't go ahead with the merge.

**Comment by oyvindberg** [comment]
Great, let's get it in then :)

you'll probably have to find the invite and accept it before you get that merge button, ping me if you cannot find it

**Pull request merged**
